mscore/workspacedialog.cpp

```diff
@@ -37,7 +37,7 @@ void MuseScore::createNewWorkspace()
 
 void MuseScore::editWorkspace()
       {
-      if (!WorkspacesManager::currentWorkspace() && !WorkspacesManager::currentWorkspace()->readOnly())
+      if (WorkspacesManager::currentWorkspace() && !WorkspacesManager::currentWorkspace()->readOnly())
```

Comment:
With this correction this check will disable editing workspace if it is not read-only while it seems like the opposite should happen. Could this check perhaps be intended to be like this?
```
if (!WorkspacesManager::currentWorkspace() || WorkspacesManager::currentWorkspace()->readOnly())
      return;
```

Reply:
Very confusing code, from the point of view of a static analyzer, there may be an appeal to an invalid pointer. Thank you, I will look what should be.

Reply:
> With this correction this check will disable editing workspace if it is not read-only while it seems like the opposite should happen. Could this check perhaps be intended to be like this?
> 
> ```
> if (!WorkspacesManager::currentWorkspace() || WorkspacesManager::currentWorkspace()->readOnly())
>       return;
> ```

Yes, that definitely seems to be the intention.

Reply:
Changed the code